pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -728,6 +723,21 @@ private BrokerResponseNative handleRequest(long requestId, String query,
     }
   }
 
+  private BrokerResponseNative getEmptyBrokerOnlyResponse(long requestId, String query,
+      RequesterIdentity requesterIdentity, RequestContext requestContext, PinotQuery pinotQuery, String tableName) {
+    if (pinotQuery.isExplain()) {
+      // EXPLAIN PLAN results to show that query is evaluated exclusively by Broker.
+      return BrokerResponseNative.BROKER_ONLY_EXPLAIN_PLAN_OUTPUT;
+    }
+
+    // Send empty response since we don't need to evaluate either offline or realtime request.
+    BrokerResponseNative brokerResponse = BrokerResponseNative.empty();
```

Comment:
Would adding **_brokerMetrics** here be useful as well ?

Reply:
adding how exactly?

Reply:
oh, I meant adding new metric to BrokerMeter to track this case, if its useful.

Reply:
ah I see. I'm leaning no for now. The initial case where we return this empty response was when the query always returned false. And that one didn't increment any metric. As a Pinot operator, I'm not sure what I would do with this metric. 

Maybe the bigger is that we have several case where we return early, but we don't actually log query.